en/advanced/healthcheck-graceful-shutdown.md

```diff
@@ -10,7 +10,18 @@ redirect_from: "/advanced/healthcheck-graceful-shutdown.html"
 
 ## Graceful shutdown
 
-When you deploy a new version of your application, you must replace the previous version. The [process manager](pm.html) you're using will first send a SIGTERM signal to the application to notify it that it will be killed. Once the application gets this signal, it will stop accepting new requests, finish all the ongoing requests, and clean up the resources it used,  including database connections and file locks.
+When you deploy a new version of your application, you must replace the previous version. The [process manager](pm.html) you're using will first send a SIGTERM signal to the application to notify it that it will be killed. Once the application gets this signal, it should stop accepting new requests, finish all the ongoing requests, clean up the resources it used,  including database connections and file locks then exit.
+
+### Example Graceful Shutdown
+```js
+process.on('SIGTERM', () => {
+  debug('SIGTERM signal received: closing server')
+  server.close(() => {
+    debug('Http server closed')
+  })
+})
+```
+_Note `yarn` may exit early if it receives `SIGTERM`._
```

Comment:
We typically do not write any yarn-specific instructions anywhere in the project. But it would be very helpful to know if this affects npm as well (I would guess it does). But even then, can you clarify what this means, exactly? It is not clear where yarn (or npm) comes into play with a running server in this documentation around the SIGTERM. Is this a warning to ensure when you send a SIGTERM to send it to the right process or something?

Reply:
Happy to remove this if you would rather. This caused me a fair bit of confusion when I was testing this which is why I added it. NPM does not show this bug. Hopefully, I have made this a bit clearer and I have put a link to a yarn gihub issue.

Reply:
Thank you. The last comment says the issue is resolved in latest yarn.

Reply:
Removed it :)